pbj-core/pbj-runtime/src/main/java/com/hedera/pbj/runtime/io/buffer/BufferedData.java

```diff
@@ -372,11 +372,13 @@ public long getBytes(final long offset, @NonNull final BufferedData dst) {
     @Override
     public Bytes getBytes(long offset, long length) {
         final var len = Math.toIntExact(length);
-        if(len < 0) throw new IllegalArgumentException("Length cannot be negative");
-        // It is vital that we always copy here, we can never assume ownership of the underlying buffer
-        final var copy = new byte[len];
-        buffer.get(Math.toIntExact(offset), copy, 0, len);
-        return Bytes.wrap(copy);
+        if (direct) {
+            final var copy = new byte[len];
+            buffer.get(Math.toIntExact(offset), copy, 0, len);
+            return Bytes.wrap(copy);
+        } else {
```

Comment:
Why are heap byte buffers handled differently? According to the (removed) comment above, the returned Bytes object should always be a copy rather than a wrapper over the existing buffer.

Reply:
I don't have a good explanation, so I rolled back the change :)